manifests/params.pp

```diff
@@ -16,272 +16,327 @@
 class snmp::params {
   # If we have a top scope variable defined, use it, otherwise fall back to a
   # hardcoded value.
-  $agentaddress = $::snmp_agentaddress ? {
-    undef   => [ 'udp:127.0.0.1:161', 'udp6:[::1]:161' ],
-    default => $::snmp_agentaddress,
+  if defined('$snmp_agentaddress') {
```

Comment:
You have right, i have done the correction on my pull request of the project puppet network but I forgot to back port it in this projet.